channel.go

```diff
@@ -425,12 +425,14 @@ type Registrar interface {
 // information.
 //
 // Register panics if the channel was constructed with an alternate root
-// handler.
+// handler that does not support Register.
 func (ch *Channel) Register(h Handler, methodName string) {
-	if _, ok := ch.handler.(channelHandler); !ok {
-		panic("can't register handler when channel configured with alternate root handler")
+	r, ok := ch.handler.(registrar)
+	if !ok {
+		panic("can't register handler when channel configured with alternate root handler without Register method")
```

Comment:
seems like this will break existing callers - is this intended? do we want to make this optional rather than mandatory?

Reply:
This should be more permissive than before, previously we panicked if the `handler` wasn't of the type `channelHandler`, but now we check for an interface (that `channelHandler` implements), so any case that worked previously should also work now. Any specific scenario you had in mind that now breaks?

Reply:
right - somehow i missed the added `Register()` method fo channelHandler. 😓  s'all good